msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -779,34 +675,54 @@ public void acquireToken(@NonNull final Activity activity,
                              @Nullable final String[] extraScopesToConsent,
                              @Nullable final String authority,
                              @NonNull final AuthenticationCallback callback) {
-        String loginHint = null;
 
-        if (null != account) {
-            loginHint = account.getUsername();
-        }
+        AcquireTokenParameters acquireTokenParameters = new AcquireTokenParameters();
+        acquireTokenParameters.setActivity(activity);
+        acquireTokenParameters.setAccount(account);
+        acquireTokenParameters.setScopes(Arrays.asList(scopes));
+        acquireTokenParameters.setUIBehavior(uiBehavior);
+        acquireTokenParameters.setExtraQueryStringParameters(extraQueryParameters);
+        acquireTokenParameters.setExtraScopesToConsent(Arrays.asList(extraScopesToConsent));
+        acquireTokenParameters.setAuthority(authority);
+        acquireTokenParameters.setAccountRecord(getAccountRecord(acquireTokenParameters.getAccount()));
+        acquireTokenParameters.setCallback(callback);
+
+        acquireToken(acquireTokenParameters);
+    }
+
+
+    /**
+     * Acquire token interactively, will pop-up webUI. Interactive flow will skip the cache lookup.
+     * Default value for {@link UiBehavior} is {@link UiBehavior#SELECT_ACCOUNT}.
+     *
+     * Convey parameters via the AquireTokenParameters object
+     *
+     * @param acquireTokenParameters
+     */
+    public void acquireToken(@NonNull AcquireTokenParameters acquireTokenParameters) {
 
-        final MSALAcquireTokenOperationParameters params =
-                getInteractiveOperationParameters(
-                        activity,
-                        scopes,
-                        loginHint,
-                        uiBehavior,
-                        extraQueryParameters,
-                        extraScopesToConsent,
-                        authority,
-                        account
-                );
+
+        acquireTokenParameters.setAccountRecord(getAccountRecord(acquireTokenParameters.getAccount()));
```

Comment:
Ooph some gymnastics going on here... Not immediately apparent how we might clean this up - perhaps we consider passing a homeAccountId and hitting the cache later?

Reply:
Discussed offline.